### ISSUE TITLE
Clixon CLI issue: when I try to print the value of the leaf node nothing appeared

### ISSUE THREAD

**Issue by vvsosed** [clicon/clixon#345]
I used command  CLI command "show system name" . "name" is the nested leaf value in the "system" container node. The right output should be in my case 
MH-T280@00000000>show system name
name 00000000;
MH-T280@00000000>
but I got nothing.

If I try to print the whole container it did fine:
MH-T280@00000000>show system
radio-bridge-tg-system:system {
    name 00000000;
    reboot-needed false;
    control {
        soft-watchdog-enabled true;
    }
    state {
        product MH-T280;
<.... other output>

I also enabled detailed logs in clixon_backend app and found log strings that can be considered RPC i/o between CLI client and clixon_backend:

clicon_msg_decode <rpc xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" username="root" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="42"><get content="all"><nc:filter nc:type="xpath" nc:select="/rb-tg-sys:system/rb-tg-sys:name" xmlns:rb-tg-sys="[http://siklu.com/yang/tg/system"/></get></rpc>](http://siklu.com/yang/tg/system%22/%3E%3C/get%3E%3C/rpc%3E)
<..... a lot of other output>
from_client_msg cbret:<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"><data><system xmlns="[http://siklu.com/yang/tg/system"><name>00000000</name></system></data></rpc-reply>](http://siklu.com/yang/tg/system%22%3E%3Cname%3E00000000%3C/name%3E%3C/system%3E%3C/data%3E%3C/rpc-reply%3E)

Based on this I supposes that maybe CLI client receives valid RPC response but not display it properly.

Also, I've found that this issue appeared after [Changed C-API for xml translation/print the internal cxobj tree data structure to other formats](https://github.com/clicon/clixon/commit/0c79298e762c7d1cc28e4bea3a67e6c0524dc04e) commit.

**Comment by olofhagsand** [comment]
Cannot recreate it, need some more info.
1) please try and see if the issue remains after the fix of https://github.com/clicon/clixon/issues/343 ? Since that fix changed the print C-API slightly.
2) Otherwise, can you provide the sub-part of the YANG of system/name as well as the CLI command with which you print system/name?

**Comment by vvsosed** [comment]
1. The Issue still remains after the autocli:hide-show fix
2. This is the YANG datamodel example at which I can observe the problem

> module issue-example {
>     yang-version "1";
> 
>     // namespace
>     namespace 'http://example.com/ns/issue';
> 
>     revision 2022-07-25 {
>         description
>           "Initial revision";
>         reference
>           "TBD";
>     }
> 
>     grouping system-group {
>         container system {
>             leaf name {
>                 type string;
>             }
>             leaf location {
>                 type string;
>             }
>         }
>     }
> 
>     uses system-group;
> }

so when I type in cli console "show system" I see output like

> name SomeName
> location SomeLocation

but when I type command "show system name" I don't see any output

**Comment by vvsosed** [comment]
I was able to fix the issue by editing [`if (clixon_txt2file(stdout, xp, 0, cligen_output, 1, 1) < 0)`](https://github.com/clicon/clixon/blob/9d65fb81d75493fb9c996b0638e9ea1e6e9eb7a1/apps/cli/cli_show.c#L777). When I define `skiptop = 0` instead of 1 in the call (i.e. if (clixon_txt2file(stdout, xp, 0, cligen_output, **0**, 1)), the issue dissapears.

**Comment by olofhagsand** [comment]
Ok, in which file/function? Is this a patch that should be committed?

**Comment by vvsosed** [comment]
This is link to place which I talk about
https://github.com/clicon/clixon/blob/9d65fb81d75493fb9c996b0638e9ea1e6e9eb7a1/apps/cli/cli_show.c#L777
so call sequence

**Comment by olofhagsand** [comment]
I am still curious on some details on this.
In my system using cli_show_auto() I get:
```
olof@alarik /> show auto system     
<system xmlns="urn:example:clixon">
   <name>foo</name>
   <location>hej</location>
</system>
olof@alarik /> show auto system name 
CLI syntax error: "show auto system name": Incomplete command
olof@alarik /> show auto system name foo 
<name>foo</name>
```
If I interpret your issue report correctly, you want to do:
```
olof@alarik /> show auto system name 
   <name>foo</name>
olof@alarik /> 
```
is that correct?
I suspect you have some other autocli setting, I remember this requirement from you earlier.

**Comment by olofhagsand** [comment]
I see you probably use `show @datamodelshow` as defined in
https://clixon-docs.readthedocs.io/en/latest/cli.html#tree-expansion

**Comment by olofhagsand** [comment]
OK, recreated it with @datamodelshow and text format. 
Patch submitted, also for JSON output.
Please verify patch.

**Comment by vvsosed** [comment]
Thanks a lot for the patch. I have verified - the issue was fixed.

In our *.cli file, we are using @datamodelstate. As I am understanding this is the same as @datamodelshow but with the state. So, your assumption was right.

I also have yet one question. I typed "show system" to print "system" container and received the output

> MH-T280@qwweerty>show system     
> radio-bridge-tg-system:system {
>     name qwweerty;
>     reboot-needed false;
>     location MyLocation;
>     ...
> }
> MH-T280@qwweerty>

Here _"radio-bridge-tg-system"_ is the module name. The problem is that our current CLI, that based on the old version of the clixon library, returns a value without a module name prefix. So my question is which variant right? Is it possible to enable or disable the appearance of the "prefix" by a configuration for consistency with previous behaviour?
